web/pdf_find_controller.js

```diff
@@ -105,11 +109,69 @@ var PDFFindController = (function PDFFindControllerClosure() {
       });
     },
 
+    // Helper for multiple search.
+    // Sorting array of objects { match: <match>, matchLength: <matchLength> }
+    // in increasing index first and then the lengths.
+    sortMatches: function PDFFindController_sortMatches(matchesWithLength) {
+      matchesWithLength.sort(function(a, b) {
+        return a.match === b.match ?
+          a.matchLength - b.matchLength : a.match - b.match;
+        });
+    },
+
+    // Helper for multiple search - for cases when one search term
+    // include another search term (for example, "tamed tame").
+    // Looking for the same elements in the 'matches' and
+    // leave elements with a longer match-length.
+    // Array matchesWithLength is pre-sorted by method sortMatches.
+    // Example - before:
+    // matches       1 2 2 2 5 7
+    // matchesLength 4 4 5 6 3 2
+    // After:
+    // matches       1 2 5 7
+    // matchesLength 4 6 3 2
+    clearMatches: function PDFFindController_clearMatches(matchesWithLength) {
```

Comment:
Is it possible to modify this function to also pass in `matches` and `matchesLength` when [calling it](https://github.com/mozilla/pdf.js/pull/5579/files#diff-485f8990604b45b959a20e90265d8044R196)?
That way you could avoid creating temporary arrays by just `push`ing directly to `matches` and `matchesLength` instead.

Also, since this function seems to assume that `matchesWithLength` is already sorted, would it make sense to just call `this.sortMatches` from within function instead of doing that before calling this function?